test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs

```diff
@@ -108,5 +115,672 @@ public void TreeEnsembleFeaturizerOutputSchemaTest()
+
+            // The second pipeline trains a tree featurizer on a bin-based normalized feature, so the second pipeline
+            // is different from the first pipeline.
+            Assert.NotEqual(metrics.MeanAbsoluteError, secondMetrics.MeanAbsoluteError);
+            Assert.NotEqual(metrics.MeanSquaredError, secondMetrics.MeanSquaredError);
+        }
+
+        [Fact]
+        public void TestFastTreeBinaryFeaturizationInPipelineWithOptionalOutputs()
+        {
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateBinaryLabelFloatFeatureVectorFloatWeightSamples(dataPointCount).ToList();
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var trainerOptions = new FastTreeBinaryTrainer.Options
+            {
+                NumberOfThreads = 1,
+                NumberOfTrees = 10,
+                NumberOfLeaves = 4,
+                MinimumExampleCountPerLeaf = 10,
+                FeatureColumnName = "Features",
+                LabelColumnName = "Label"
+            };
+
+            var options = new FastTreeBinaryFeaturizationEstimator.Options()
+            {
+                InputColumnName = "Features",
+                TrainerOptions = trainerOptions,
+                TreesColumnName = null,
+                PathsColumnName = null,
+                LeavesColumnName = "Leaves"
+            };
+
+
+            bool isWrong = false;
+            try
+            {
+                var wrongPipeline = ML.Transforms.FeaturizeByFastTreeBinary(options).
+                    Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Trees", "Leaves", "Paths")).
+                    Append(ML.BinaryClassification.Trainers.SdcaLogisticRegression("Label", "CombinedFeatures"));
+                var wrongModel = wrongPipeline.Fit(dataView);
+            }
+            catch
+            {
+                isWrong = true; // Only "Leaves" is produced by tree featurizer, so accessing "Trees" and "Paths" may lead to an error.
+            }
+            Assert.True(isWrong);
+
+            var pipeline = ML.Transforms.FeaturizeByFastTreeBinary(options).
+                Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Leaves")).
+                Append(ML.BinaryClassification.Trainers.SdcaLogisticRegression("Label", "CombinedFeatures"));
+            var model = pipeline.Fit(dataView);
+            var prediction = model.Transform(dataView);
+            var metrics = ML.BinaryClassification.Evaluate(prediction);
+
+            Assert.True(metrics.Accuracy > 0.98);
+            Assert.True(metrics.LogLoss < 0.05);
+            Assert.True(metrics.AreaUnderPrecisionRecallCurve > 0.98);
+        }
+
+        [Fact]
+        public void TreeEnsembleFeaturizingPipelineMulticlass()
+        {
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(dataPointCount).ToList();
```

Comment:
For a multi-class UnitTest, I think we'd want to create a multi-class dataset:
```suggestion
            var data = SamplesUtils.DatasetUtils.GenerateRandomMulticlassClassificationExamples(dataPointCount).ToList();
```

Otherwise, it's creating a regression dataset.